apisix/core/resolver.lua

```diff
@@ -19,15 +19,31 @@
 --
 -- @module core.resolver
 
-local json = require("apisix.core.json")
-local log = require("apisix.core.log")
-local utils = require("apisix.core.utils")
+local json           = require("apisix.core.json")
+local log            = require("apisix.core.log")
+local utils          = require("apisix.core.utils")
+local dns_utils      = require("resty.dns.utils")
+
+
+local HOSTS_IP_MATCH_CACHE = {}
 
 
 local _M = {}
 
 
+local function init_hosts_ip()
+    local hosts, err = dns_utils.parseHosts()
+    if not hosts then
+        return hosts, err
+    end
+    HOSTS_IP_MATCH_CACHE = hosts
```

Comment:
When will `HOSTS_IP_MATCH_CACHE` be cleaned? APISIX will not update it after it is read at startup. Unless APISIX is restarted.

Reply:
This kind of configuration is basically unchanged, so I didn't consider reloading logic at that time